ModuleSystem/module_scripts_ai.py

```diff
@@ -2795,34 +2795,66 @@
 #White Mountains mountaintops, far apart, and update the values below with them
 tld_white_mountains_p1=( 26,-1)
 tld_white_mountains_p2=(-44,24)
+tld_white_mountains_min_y=-1200   # An Y*100 coordinate always North of the WM - I'm using Hornburg Y right now
 #And let me handle the rest obtaining the line equation y=mx+n
 tld_white_mountains_m=(tld_white_mountains_p1[1]-tld_white_mountains_p2[1])*1.0/(tld_white_mountains_p1[0]-tld_white_mountains_p2[0])
 tld_white_mountains_n=tld_white_mountains_p1[1]-tld_white_mountains_m*tld_white_mountains_p1[0]
 tld_white_mountains_m=int(round(tld_white_mountains_m*100,0))
 tld_white_mountains_n=int(round(tld_white_mountains_n*10000,0)) # When doing y=x*m + n, both x and m are mult * 100, k now must be mult an extra *100

```

Comment:
Sorry! I hadn't see your replies until now.

Yes, that's exacly what it does. The old code was at least partially working incorrectly, and that was the motive of a lot of missions send the player on errands to the other freaking side of the White Mountains. I didn't understand exacly what they were doing at the final check, so I opted to implement this partially less obscure method in a way that can be easily modified by a mapper, that might not have his maths so fresh.

It certainly could have an early bailout to the south too (that went over my head, lol), and it would speed up a little bit most of the southern cases too. Minas Tirith Y is a good candidate to that point.

Will do it, and try to add some comments.